presto-main/src/main/java/io/prestosql/sql/planner/EffectivePredicateExtractor.java

```diff
@@ -218,8 +222,11 @@ public Expression visitTableScan(TableScanNode node, Void context)
         {
             Map<ColumnHandle, Symbol> assignments = ImmutableBiMap.copyOf(node.getAssignments()).inverse();
 
-            // TODO: replace with metadata.getTableProperties() when table layouts are fully removed
```

Comment:
The TODO: should be retained. The idea is to eventually get rid of `getEnforcedConstraint()` entirely. The TableScan node shouldn't be tracking what constraint has been pushed so far or what the "effective predicate" of the table scan is. That's the purpose of the properties exposed by TableProperties.

Reply:
Not if we can manage it. We'll have to figure out a way to deal with it once we have an exploratory optimizer with no clear "phases"